src/common/array.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <vector>
+
+namespace vx
+{
+template<typename T>
+class array
+{
+  public:
+    explicit array() {}
+    explicit array(int size) : backend(size) {}
+    inline T& add()
```

Comment:
Just some fun C++ ramblings here -- I was about to remark that the inline keyword is useless here, since template-parametrized methods can be defined inline without the inline keyword. But then I checked: even though the method is defined inline doesn't necessarily mean that it will be inlined by the compiler 😆 (although this code is mostly trivial so it most likely will)